backend/src/requests/user-content-request/user-content-request.service.ts

```diff
@@ -402,4 +409,63 @@ export class UserContentRequestService {
       throw new BadRequestException(e);
     }
   }
+
+  /**
+   * set vote for one userContentId for one user,
+   * removes the vote if the parameter vote is set to 'none'
+   * @param vote
+   * @param userContentId
+   * @param userId
+   *
+   * @throws NotFoundException
+   */
+  async updateUserVote(
+    vote: VoteDto,
+    userContentId: string,
+    userId: string,
+  ): Promise<Vote | null> {
+    // check question exists
+    const author =
+      await this.userContentService.getAuthorOfUserContent(userContentId);
+    if (author == null) {
+      throw new NotFoundException(
+        'User content ' + userContentId + " doesn't exist!",
+      );
+    }
+
+    const oldVote = await this.voteService.getVote(userContentId, userId);
+    if (!!oldVote) {
```

Comment:
This should be changed to `if (oldVote)` because !!oldVote is the same as oldVote

Reply:
Fixed within 94dc50f